collector/cache.go

```diff
@@ -0,0 +1,453 @@
+package collector
+
+import (
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+func init() {
+	Factories["cache"] = NewCacheCollector
```

Comment:
This is my fault for missing to review this for so long, but we've made some refactorings which has changed how the registration is done. Please have a look at the new `registerCollector` function we use elsewhere